src/main/js/PullRequestDetails.tsx

```diff
@@ -166,341 +157,245 @@ const IgnoredMergeObstacles = styled.div`
   border-bottom: 1px solid hsla(0, 0%, 85.9%, 0.5);
 `;
 
-class PullRequestDetails extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      ...this.state,
-      loadingDryRun: false,
-      mergeButtonLoading: true,
-      rejectButtonLoading: false,
-      targetBranchDeleted: false
-    };
-  }
-
-  componentDidMount(): void {
-    const { pullRequest } = this.props;
-    this.getMergeDryRun(pullRequest);
-  }
-
-  shouldRunDryMerge = (pullRequest: PullRequest) => {
-    return (
-      pullRequest._links.mergeCheck && (pullRequest._links.mergeCheck as Link).href && pullRequest.status === "OPEN"
-    );
-  };
+const PullRequestDetails: FC<Props> = ({ repository, pullRequest }) => {
+  const [t] = useTranslation("plugins");
+  const [targetBranchDeleted, setTargetBranchDeleted] = useState(false);
 
-  onMergeModalClosed = () => {
-    this.setState({
-      mergeCheck: undefined
-    });
-  };
+  const { reject, isLoading: rejectLoading, error: rejectError } = useRejectPullRequest(repository, pullRequest);
+  const { merge, isLoading: mergeLoading, error: mergeError } = useMergePullRequest(repository, pullRequest);
+  const { data: mergeCheck, isLoading: mergeDryRunLoading, error: mergeDryRunError } = useMergeDryRun(
+    repository,
+    pullRequest,
+    (targetDeleted: boolean) => setTargetBranchDeleted(targetDeleted)
+  );
 
-  getMergeDryRun(pullRequest: PullRequest) {
-    if (this.shouldRunDryMerge(pullRequest)) {
-      check(pullRequest)
-        .then(response => {
-          this.setState({
-            mergeCheck: response,
-            targetBranchDeleted: false,
-            loadingDryRun: false,
-            mergeButtonLoading: false
-          });
-        })
-        .catch(err => {
-          if (err instanceof NotFoundError) {
-            this.setState({
-              mergeButtonLoading: false,
-              loadingDryRun: false,
-              targetBranchDeleted: true
-            });
-          } else {
-            this.setState({
-              error: err,
-              loadingDryRun: false,
-              mergeButtonLoading: false
-            });
-          }
-        });
-    }
-  }
+  const error = rejectError || mergeError || mergeDryRunError;
 
-  findStrategyLink = (links: Link[], strategy: string) => {
+  const findStrategyLink = (links: Link[], strategy: string) => {
     return links?.filter(link => link.name === strategy)[0].href;
   };
```

Comment:
I'm not sure if the whole function would be better placed inside the merge hook.

Reply:
Maybe, but this seems like another bigger refactoring. Could we focus on react-query first?